client/packages/invoices/src/Prescriptions/LineEditView/PrescriptionLineEditForm.tsx

```diff
@@ -237,25 +237,25 @@ export const PrescriptionLineEditForm: React.FC<
 
   const key = item?.id ?? 'new';
 
-  // //MOCK DATA FOR TESTING ABBREVIATIONS
-
-  const defaultDirections = [
-    { id: '1', item_link_id: 'a', direction: '1 per day', priority: 2 },
-    { id: '2', item_link_id: 'b', direction: '2 per day', priority: 3 },
-    { id: '3', item_link_id: 'c', direction: '3 per day', priority: 1 },
-  ];
-
-  // //END OF MOCK DATA
-
   const { data: options = [] } = useAbbreviations();
 
   const saveAbbreviation = () => {
     if (!abbreviation) return;
-
     const note = getPrescriptionDirections(abbreviation, options);
     updateNotes(note);
+    setDefaultDirection('');
   };
 
+  const saveDefaultDirection = useCallback(
+    (direction: string) => {
+      setDefaultDirection(direction);
+      const note = getPrescriptionDirections(direction, options);
+      updateNotes(note);
+      setAbbreviation('');
+    },
+    [defaultDirection]
```

Comment:
I'm not sure why we need this dependency but I haven't used useCallback much myself...

Reply:
Actually I'm getting suspicious of it... yea I tested just removing `useCallback` and now the state isn't cleared and all is well... yup so you actually, shockingly, have a dependency on `updateNotes` the function as it is memoized using `useCallback` as well so in this function you were calling an old version that hadn't got re-evaluated with the draft lines in it. Yikes, it's a minefield out there...

I'd honestly be happy with just not wrapping this one in useCallback, surely dropdown menu isn't too expensive to rerender? Then again, this situation exists maybe I shouldn't be too optimistic hahahaha.